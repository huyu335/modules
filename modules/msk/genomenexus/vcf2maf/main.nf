process GENOMENEXUS_VCF2MAF {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/genomenexus:latest':
        'ghcr.io/msk-access/genomenexus:latest' }"

    input:
    tuple val(meta), path(vcf)

    output:
    tuple val(meta), path("vcf2maf_output/${meta.id}.maf"), emit: maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    

    """
    python3 /annotation-tools/vcf2maf.py -i ${vcf} ${args}
    sleep 3

    echo '"${task.process}": vcf2maf.py --help' > versions.yml
    """
    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """

    mkdir vcf2maf_output
    touch ${meta.id}.maf
    cp ${meta.id}.maf vcf2maf_output/
    sleep 3


    echo '"${task.process}": vcf2maf.py --help' > versions.yml
    """
}
