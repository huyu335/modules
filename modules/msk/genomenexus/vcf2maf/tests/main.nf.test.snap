{
    "vcf2maf - test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,548723a34b74a614b654bc8abff855eb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dc4c88b6cebf51af905b5f4d5fa16152"
                ],
                "maf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,548723a34b74a614b654bc8abff855eb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dc4c88b6cebf51af905b5f4d5fa16152"
                ]
            }
        ],
        "timestamp": "2024-04-03T13:53:02.512965"
    },
    "vcf2maf - test - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dc4c88b6cebf51af905b5f4d5fa16152"
                ],
                "maf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dc4c88b6cebf51af905b5f4d5fa16152"
                ]
            }
        ],
        "timestamp": "2024-04-03T13:53:05.988213"
    }
}